### PR TITLE
TCH-37: add cache with redis to setup

### PR DESCRIPTION
**Pull request by imwillim**

### CODE REVIEW

File: spec/lib/cache_manager_spec.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe CacheManager do
+  include_context 'redis mock'
```
[the remark this answers is not in the document]
Comment (imwillim) [reply spec/lib/cache_manager_spec.rb]: Config is set for this `include_context`

File: lib/cache_manager.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class CacheManager
+  def self.fetch_value(key)
+    RedisWrapper.redis_pool.then do |redis|
+      redis.get(key)
+    end
+  end
+
+  def self.assign_value(key, value)
```
[the remark this answers is not in the document]
Comment (imwillim) [reply lib/cache_manager.rb:10]: Pointed, but will not be done in this PR

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -63,6 +63,8 @@
   config.filter_rails_from_backtrace!
   # arbitrary gems may also be filtered via:
   # config.filter_gems_from_backtrace("gem name")
+
+  config.include_context 'redis mock'
```
Comment (bmhuyquoc104) [inline spec/rails_helper.rb:67]: will this autoload the file that not use this context ?

File: spec/support/redis_mock.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+RSpec.shared_context 'redis mock' do
+  let(:redis) { MockRedis.new }
+
+  before(:each) do
```
Comment (bmhuyquoc104) [inline spec/support/redis_mock.rb:6]: is before(:each) already a default behavior ?

**Review by bmhuyquoc104 — APPROVED** [review]

**Pull request merged**